aiohttp/base_protocol.py

```diff
@@ -54,6 +57,10 @@ def connection_made(self, transport: asyncio.BaseTransport) -> None:
 
     def connection_lost(self, exc: Optional[BaseException]) -> None:
         self._connection_lost = True
+        if exc is not None:
```

Comment:
Please don't touch this file but add all required logic to `client_proto.py`

Reply:
ok